var/spack/repos/builtin/packages/geopm/package.py

```diff
@@ -45,24 +36,20 @@ class Geopm(AutotoolsPackage):
     variant('doc', default=True, description='Create man pages with ruby-ronn.')
     variant('openmp', default=True, description='Build with OpenMP.')
     variant('ompt', default=False, description='Use OpenMP Tools Interface.')
-    variant('hwloc', default=False, description='Build with hwloc, deprecated and ignored after v0.5.1.')
     variant('gnu-ld', default=False, description='Assume C compiler uses gnu-ld.')
 
     # Added dependencies.
-    depends_on('m4', type='build')
-    depends_on('automake', type='build')
-    depends_on('autoconf', type='build')
-    depends_on('libtool', type='build')
```

Comment:
These dependencies are very likely needed when you build from `master` or `develop`, due to how autotools are used. Can you please double check if the correct thing to do is instead change the lines above to:
```python
depends_on('m4', type='build', when='@develop,master')
...
```
?

Reply:
Thanks for the hint. I tried this yesterday and still have to work out how to best run autogen.sh when building from develop. I assume this is the right path: https://github.com/spack/spack/blob/42633e34ab0f7db7df5dc79aabfa63df31dd47a6/var/spack/repos/builtin/packages/libuv/package.py ?